### PR TITLE
Load a demo case when demo app is launched

### LINKED ISSUE #32: Demo data in docker-compose
We could provide an example with real data by following [this tutorial](https://github.com/38/d4-format/blob/master/pyd4/examples/D4%20Tutorial%20in%20Python.ipynb). There are also real files that can be used!

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Loads a demo case when demo app is launched (fix #32)

### How to test:
- Launch demo app
- Use cases endpoint to make sure demo case is there

### Review:
- [x] Code approved by HS
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: src/chanjo2/populate_demo.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from chanjo2.crud.samples import create_sample_in_case
+from chanjo2.dbutil import get_session
+from chanjo2.demo import d4_demo_path
+from chanjo2.models.pydantic_models import CaseCreate, SampleCreate
+from chanjo2.models.sql_models import Case
+from sqlalchemy.orm import sessionmaker
+
+db: sessionmaker = next(get_session())
+
+DEMO_CASE = {"name": "internal_id", "display_name": "643594"}
```
Comment (northwestwitch) [inline src/chanjo2/populate_demo.py:11]: Same as in Scout demo, for testing purposes, but can be any case

**Comment by northwestwitch** [comment]
Local test launching the app: 🆗 

image

The case is found in database: 🆗 

image

File: src/chanjo2/populate_demo.py (hunk 1 above)
Comment (henrikstranneheim) [inline src/chanjo2/populate_demo.py:11]:
```suggestion
DEMO_CASE: Dict[str, str] = {"name": "internal_id", "display_name": "643594"}
```

File: src/chanjo2/populate_demo.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+DEMO_SAMPLE = {
```
Comment (henrikstranneheim) [inline src/chanjo2/populate_demo.py:12]: Add  type hint

File: src/chanjo2/populate_demo.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def load_demo_data() -> Case:
+    """Loads demo data into the database of a demo instance of Chanjo2."""
+    load_demo_case()
+    load_demo_sample()
+    return get_case(db=db, case_name=DEMO_CASE["name"])
+
+
+def load_demo_case():
```
Comment (henrikstranneheim) [inline src/chanjo2/populate_demo.py:27]:
```suggestion
def load_demo_case() -> None:
```

File: src/chanjo2/populate_demo.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    """Load a demo case into the database."""
+    case: CaseCreate = CaseCreate(**DEMO_CASE)
+    create_db_case(db=db, case=case)
+
+
+def load_demo_sample():
```
Comment (henrikstranneheim) [inline src/chanjo2/populate_demo.py:33]:
```suggestion
def load_demo_sample() -> None:
```

File: tests/src/chanjo2/test_populate_demo.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+from chanjo2.models.sql_models import Case
+from fastapi.testclient import TestClient
+from sqlalchemy.orm import sessionmaker
+
+client = TestClient(app)
+db: sessionmaker = next(get_session())
+
+
+def test_load_demo_data():
+    """Test the function that adds demo data."""
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/test_populate_demo.py:13]:
```suggestion
    """Test loading demo data."""
```

File: tests/src/chanjo2/test_populate_demo.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    # WHEN the app is launched
+    with TestClient(app) as client:
+        # THEN a demo case should be found in the database
+        result: list = get_cases(db)
+        assert isinstance(result[0], Case)
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/test_populate_demo.py:19]:
```suggestion
        cases: List[Case] = get_cases(db)
        assert isinstance(cases[0], Case)
```

**Review by henrikstranneheim — APPROVED** [review]
👍

File: src/chanjo2/populate_demo.py (hunk 2 above)
Comment (northwestwitch) [reply src/chanjo2/populate_demo.py]: Done, thanks!

**Pull request merged**